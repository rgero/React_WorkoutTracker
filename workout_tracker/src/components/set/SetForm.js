import React, {useState} from 'react';

import '../../styles/components/inputs.css'

export const SetForm = ({set = {}, onSubmit})=> 
{
    const [reps, setReps] = useState(set.reps ? set.reps : "")
    const [weight, setWeight] = useState(set.weight ? set.weight : "");
    const [error, setError] = useState("")

    const processChange = (event, callback) => {
        const amount = event.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            callback(amount);
        }
    }

    const processNewSet = (e) => {
        e.preventDefault();
        
        if (reps === "" || weight === "")
        {
            setError("Reps and Weight need to be populated");
            return;
        }

        let newSet = { reps, weight}
        setReps("");
        setWeight("");
        setError("");
        onSubmit(newSet);
    }

    return (
        <form className="form" onSubmit={processNewSet}>
            {error !== '' && <p className="form__error">{error}</p>}

            {/* Reps */}
            <div className="input-entry">
                <label>Reps</label>
                <input
                    type="text-input"
                    className="text-input"
                    placeholder="Reps"
                    autoFocus
                    value={reps}
                    onChange={e => processChange(e, setReps)}
                />
            </div>

            <div className="input-entry">
                <label>Weight</label>
                <input
                    type="text-input"
                    className="text-input"
                    placeholder="Weight"
                    value={weight}
                    onChange={e => processChange(e, setWeight)}
                />
            </div>

            <div>
                <button className="button">Save Set</button>
            </div>

        </form>
    )
}