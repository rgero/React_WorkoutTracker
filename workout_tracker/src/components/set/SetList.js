import React from 'react';

import ListGroup from 'react-bootstrap/ListGroup';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import Table from 'react-bootstrap/Table';

export const SetList = ({setList = [], onDelete=null})=> 
{
  const processDelete = (index) => {
    if (!onDelete)
    {
      return;
    }
    onDelete(index);
  }

  if (setList.length === 0)
  {
    return (null);
  }

  return (
    <Container className="pt-3" style={{width:"60%", margin:"auto", textAlign:"center" }} >
      <div style={{textAlign:"left"}}>Set List</div>
      <Table striped bordered hover size="sm">
        <thead>
          <tr>
            <th>Reps</th>
            { setList[0].weight ? (
                  <th>Weight</th>
                ) : (null) 
          }
          </tr>
        </thead>
        <tbody>
            { setList.map((set, index) => (
              <tr onClick={(e) => processDelete(index)} key={index}>
                <td>
                  {set.reps}
                </td>
                { set.weight ? (
                  <td>
                    {set.weight}
                  </td>
                ) : (null) }
              </tr>
            ))}
        </tbody>
      </Table>
    </Container>
  )
}