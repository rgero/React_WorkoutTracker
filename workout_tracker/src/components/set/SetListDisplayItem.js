import React from 'react';

export const SetListDisplayItem = ({set})=> 
{
    return (
        <div className="setListItem">
            <div>
                {set.reps}
            </div>
            <div>
                {set.weight}
            </div>
        </div>
    )
}