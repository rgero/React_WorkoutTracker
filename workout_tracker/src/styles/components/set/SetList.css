.setListItem {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

.setListHeader
{
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    font-weight: bold;
}

.setList
{
    margin: 1.2rem 0 1.2rem 0;
    border: 1px black solid;
    border-radius: 10px;
}