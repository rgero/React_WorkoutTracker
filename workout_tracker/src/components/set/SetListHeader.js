import React from 'react';

import '../../styles/components/set/SetList.css';

export const SetListHeader = ({reps, weight})=> 
{
    return (
        <div className="setListHeader">
            <div>
                Reps
            </div>
            <div>
                Weight
            </div>
        </div>
    )
}