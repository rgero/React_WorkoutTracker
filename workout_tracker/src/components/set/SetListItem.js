import React, {useState} from 'react';

import '../../styles/components/set/SetList.css';

export const SetListItem = ({set, onDelete})=> 
{
    const [color, setColor] = useState({background: "#ffffff"});

    const highlightColor = {background: "#03a9fc"}
    const unhighlightColor = {background: "#ffffff"}

    const onMouseOver = (e) => {
        setColor(highlightColor);
    }
    
    const onMouseOut = (e) => {
        setColor(unhighlightColor);
    }

    const itemClicked = (e) => {
        onDelete(set.id);
    }

    return (
        <div className="setListItem" style={color} onMouseEnter={onMouseOver.bind(this)} onMouseLeave={onMouseOut.bind(this)} onClick={itemClicked}>
            <div>
                {set.reps}
            </div>
            <div>
                {set.weight}
            </div>
        </div>
    )
}