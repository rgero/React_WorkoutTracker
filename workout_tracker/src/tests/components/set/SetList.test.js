import TestRenderer from 'react-test-renderer';
import {SetList} from '../../../components/set/SetList';

describe("Render tests", ()=> {

    test("No Data, matches snapshot", ()=> {
        let element = new TestRenderer.create(
            <SetList />
        )
        expect(element.toJSON()).toMatchSnapshot();
    })

    test("Has Data, matches snapshot", ()=> {

        let setList = [
            {
                reps: "20",
                weight: "225"
            },
            {
                reps: "100",
                weight: "315"
            }
        ]

        let element = new TestRenderer.create(
            <SetList setList={setList}/>
        )
        expect(element.toJSON()).toMatchSnapshot();
    })

})