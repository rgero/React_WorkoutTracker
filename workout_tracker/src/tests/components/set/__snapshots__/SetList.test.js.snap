// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render tests Has Data, matches snapshot 1`] = `
<div
  className="pt-3 container"
  style={
    Object {
      "margin": "auto",
      "textAlign": "center",
      "width": "60%",
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    Set List
  </div>
  <table
    className="table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Reps
        </th>
        <th>
          Weight
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        onClick={[Function]}
      >
        <td>
          20
        </td>
        <td>
          225
        </td>
      </tr>
      <tr
        onClick={[Function]}
      >
        <td>
          100
        </td>
        <td>
          315
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Render tests No Data, matches snapshot 1`] = `null`;
